### PR TITLE
Implement updated purl mappings.

### LINKED ISSUE #2155: Implement updated purl mappings
See changes to location_spec in #2154.

### PR DESCRIPTION
**Pull request by justinlittman**
closes #2155

## Why was this change made?
Nothing is easy.

## How was this change tested?
Unit, sdr-deploy

## Which documentation and/or configurations were updated?
NA

### CODE REVIEW

**Comment by justinlittman** [comment]
Before:
```
Status (n=100000):
  Success:   98358 (98.358%)
  Different: 1132 (1.132%)
  To Cocina error:     0 (0.0%)
  To Fedora error:     0 (0.0%)
  Missing:     510 (0.51%)
```

After:
```
Status (n=100000):
  Success:   98401 (98.401%)
  Different: 1089 (1.089%)
  To Cocina error:     0 (0.0%)
  To Fedora error:     0 (0.0%)
  Missing:     510 (0.51%)
```

File: app/services/cocina/from_fedora/descriptive/descriptive_builder.rb [hunk 1]
```diff
@@ -35,16 +36,16 @@ def initialize(notifier:, title_builder: Titles)
...
         # @return [Hash] a hash that can be mapped to a cocina descriptive model
-        def build(resource_element:, require_title: true)
+        def build(resource_element:, purl: nil, require_title: true)
           cocina_description = {}
           title_result = @title_builder.build(resource_element: resource_element, require_title: require_title, notifier: notifier)
           cocina_description[:title] = title_result if title_result.present?
 
-          purl_node = Purl.primary_purl_node(resource_element)
-          cocina_description[:purl] = purl_node.text if purl_node
+          purl_value = purl || Purl.primary_purl_node(resource_element, purl)&.content&.presence
```
Comment (justinlittman) [inline app/services/cocina/from_fedora/descriptive/descriptive_builder.rb:44]: This is a significant change. Using purl as provided preferred over purl as found.

File: spec/requests/create_collection_spec.rb [hunk 2]
```diff
@@ -37,7 +43,7 @@
         {"type":"http://cocina.sul.stanford.edu/models/collection.jsonld",
           "label":"#{label}","version":1,"access":{},
           "administrative":{"releaseTags":[],"hasAdminPolicy":"druid:dd999df4567","partOfProject":"Hydrus"},
-          "description":{"title":[{"value":"#{title}"}]}}
+          "description":{"title":[{"value":"#{title}"}],"purl":"http://purl.stanford.edu/gg777gg7777","access":{"digitalRepository":[{"value":"Stanford Digital Repository"}]}}}
```
Comment (justinlittman) [inline spec/requests/create_collection_spec.rb:46]: Now when there is a purl provided, `access` is populated. This is the cause of most of the test changes below.

**Review by jcoyne — APPROVED** [review]
Looks good to me. What do you think @arcadiafalcone?

**Review by arcadiafalcone — APPROVED** [review]

**Pull request merged**